src/Services/BookingService/BookingService.cs

```diff
@@ -179,16 +179,16 @@ private async Task<BoookingNextAvailabilityEntity> RetrieveNextAvailability(IEle
         private async Task<ProcessRequestEntity> ProcessDateAndTime(Dictionary<string, dynamic> viewModel, Page currentPage, FormSchema baseForm, string guid, string path)
         {
             var bookingElement = (Booking) currentPage.Elements.First(_ => _.Type.Equals(EElementType.Booking));
+            var cachedAnswers = _distributedCache.GetString(guid);
 
-            if (!currentPage.IsValid)
-            {
-                var cachedAnswers = _distributedCache.GetString(guid);
+            var convertedAnswers = cachedAnswers == null
+                ? new FormAnswers { Pages = new List<PageAnswers>() }
+                : JsonConvert.DeserializeObject<FormAnswers>(cachedAnswers);
 
-                var convertedAnswers = cachedAnswers == null
-                    ? new FormAnswers { Pages = new List<PageAnswers>() }
-                    : JsonConvert.DeserializeObject<FormAnswers>(cachedAnswers);
+            var cachedBookingInformation = JsonConvert.DeserializeObject<BookingInformation>(convertedAnswers.FormData[$"{bookingElement.Properties.QuestionId}{BookingConstants.APPOINTMENT_TYPE_SEARCH_RESULTS}"].ToString());
 
-                var cachedBookingInformation = JsonConvert.DeserializeObject<BookingInformation>(convertedAnswers.FormData[$"{bookingElement.Properties.QuestionId}{BookingConstants.APPOINTMENT_TYPE_SEARCH_RESULTS}"].ToString());
+            if (!currentPage.IsValid)
+            {
```

Comment:
cachedAnswers only requried when page is invalid, call can be moved back inside page.isValid check. No need to fetch answers if not required.